### PR TITLE
add open-iconic@1.1.1 w/ git auto-update

### LINKED ISSUE #9222: [Request] Add open-iconic
**Library name:** open-iconic
**Git repository url:** https://github.com/iconic/open-iconic
https://www.npmjs.com/package/open-iconic

### PR DESCRIPTION
**Pull request by kennynaoh**
close #9222, cc @somerandomdude

Pull request for issue: #9222 
Related issue(s): # #9222 
@x09326 Please help me review this PR, thanks!
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/iconic/open-iconic
 * Official website (optional, not the repository): https://useiconic.com/open
 * NPM package url (optional):
 * License and its reference: MIT, OFL-1.1, https://github.com/iconic/open-iconic/blob/master/ICON-LICENSE, https://github.com/iconic/open-iconic/blob/master/FONT-LICENSE
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 1810
   - Count of watchers: 93
   - Count of forks: 165
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/open-iconic/package.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "email": "yourfriends@useiconic.com",
+    "web": "http://useiconic.com/"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/iconic/open-iconic.git"
+  },
+  "licenses": [
+    "MIT",
+    "OFL 1.1"
```
Comment (x09326) [inline ajax/libs/open-iconic/package.json]: @kennynaoh Please modify it to `"OFL-1.1"`, ref: https://github.com/cdnjs/cdnjs/blob/master/tools/license-list.json#L242

**Comment by kennynaoh** [comment]
@x09326 I have fixed it. Please help me review it again. Thanks!

**Comment by x09326** [comment]
LGTM
ping @PeterDaveHello

**Comment by kennynaoh** [comment]
@x09326 i have added svg files. Please help me review it again. Thanks!

**Comment by x09326** [comment]
LGTM
ping @PeterDaveHello

**Comment by PeterDaveHello** [comment]
@pvnr0082t @maruilian11 please also take a look

File: ajax/libs/open-iconic/package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "sprite",
+    "font",
+    "png",
+    "webp"
+  ],
+  "homepage": "https://useiconic.com/open",
+  "author": {
+    "name": "Iconic",
+    "email": "yourfriends@useiconic.com",
+    "web": "http://useiconic.com/"
```
Comment (PeterDaveHello) [inline ajax/libs/open-iconic/package.json]: As we don't really use this info yet, I'm fine with it.

File: ajax/libs/open-iconic/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    "OFL-1.1"
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/iconic/open-iconic.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "font/**/!(*.less|*.scss|*.styl)",
```
Comment (pvnr0082t) [inline ajax/libs/open-iconic/package.json]: `"font/**/*.!(less|scss|styl)"` will be more clear.

File: ajax/libs/open-iconic/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          "sprite/+(open-iconic|sprite).!(*html*)",
+          "svg/*.svg"
```
Comment (pvnr0082t) [inline ajax/libs/open-iconic/package.json]: We can combine these two into `"+(svg|sprite)/*.svg"`.

**Comment by kennynaoh** [comment]
Thanks. I have updated the package.json. Please help me review it again.

**Comment by x09326** [comment]
LGTM
ping @PeterDaveHello

**Comment by kennynaoh** [comment]
@x09326 @pvnr0082t I have added `png` and `webp`, please help me review it again. Thanks!

File: ajax/libs/open-iconic/package.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/iconic/open-iconic.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "font/**/*.!(less|scss|styl)",
+          "+(svg|sprite)/*.svg",
+          "+(png|webp)/*.+(png|webp)"
```
Comment (pvnr0082t) [inline ajax/libs/open-iconic/package.json]: `png` and `webp` are under different directory and they are different file extension. I don't think it's good to combine these two.

Comment (PeterDaveHello) [inline ajax/libs/open-iconic/package.json]: I think we can just split both line 38 and line 39 out, unless we're merging them all.

**Comment by kennynaoh** [comment]
@x09326 @pvnr0082t I have split them out. Please help me review it again. Thanks!

**Comment by PeterDaveHello** [comment]
You use `**/` in the font path, then why not also use it on the others? Let's also do it on the other rules.

**Comment by kennynaoh** [comment]
@x09326 @pvnr0082t I have add `**/` in other patterns. Please help me review it. Thanks!

**Pull request merged**